brave/brave-core
# Refactors tip flows

Resolves https://github.com/brave/brave-browser/issues/3953

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:
plan 1
- clean profile
- enable rewards
- claim grant
- open two windows (one rewards page, one verified site)
- set recurring tip and make sure that rewards page reflects that

plan 2
- do plan 1
- open panel and go to site banner
- on rewards page remove recurring tip
- make sure that site banner shows checkbox again

plan 3
- clean profile
- enable rewards
- claim grant
- open two windows (one rewards page, one verified site)
- tip a site and make sure that rewards page reflects that

## Reviewer Checklist:
